platform/linux-generic/odp_queue.c

```diff
@@ -263,7 +275,7 @@ static int queue_destroy(odp_queue_t handle)
 		ODP_ERR("queue \"%s\" already destroyed\n", queue->s.name);
 		return -1;
 	}
-	if (queue->s.head != NULL) {
+	if (ring_st_is_empty(&queue->s.ring_st) == 0) {
```

Comment:
`if (!ring_st_is_empty(&queue->s.ring_st))` seems more natural here.

Reply:
I prefer style where '== 0' is used instead of '!'. Especially, when the if clause is complex enough for reader to miss the '!' sign.

Reply:
Elsewhere you write `if (ring_st_is_empty(...))`, not `if (ring_st_is_empty(...) == 1)` so this is inconsistent.

Reply:
```
if (ring_st_is_empty(&queue->s.ring_st))

if (ring_st_is_empty(&queue->s.ring_st))

if (ring_st_is_empty(&queue->s.ring_st) == 0)

if (ring_st_is_empty(&queue->s.ring_st))

if (ring_st_is_empty(&queue->s.ring_st))

if (!ring_st_is_empty(&queue->s.ring_st))

if (ring_st_is_empty(&queue->s.ring_st))

if (ring_st_is_empty(&queue->s.ring_st))
```

Find if clause for "ring is not empty" from picture above.